examples/bubbling.rs

```diff
@@ -0,0 +1,90 @@
+use bevy::prelude::*;
```

Comment:
This example needs some long-form comments (maybe at the top) explaining exactly what's going on to avoid magic.

Reply:
I've added a bunch of comments, as well as a more advanced `generic_events` example. Let me know what you think.

Reply:
Also, I'm not sure if the generic events are a feature looking for a problem, or if this is genuinely useful.